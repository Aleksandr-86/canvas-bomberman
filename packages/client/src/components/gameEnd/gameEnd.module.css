.gameEnd {
  display: flex;
  flex-direction: column;
  gap: 50px;
  width: 1280px;
  height: 720px;
  justify-content: center;
  align-items: center;
  color: #fff;
}

.gameEndText {
  font-size: 50px;
  line-height: 1.2em;
}

.gameEndButtons {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.gameEndButton {
  border-color: #fff;
  background-color: transparent;
  color: #fff;
  text-align: center;
}
