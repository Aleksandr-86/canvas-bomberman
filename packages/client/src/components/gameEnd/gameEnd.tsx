import React from 'react'
import { useAppSelector } from '../../store/hooks'
import { Link } from 'react-router-dom'
import { getCurrentScore } from '../../store/selectors'
import { Button } from '../button/button'
import styles from './gameEnd.module.css'

interface Props {
  restartGame: () => void
}

export const GameEnd: React.FC<Props> = ({ restartGame }) => {
  const currentScore = useAppSelector(getCurrentScore)

  return (
    <div className={styles.gameEnd}>
      <p className={styles.gameEndText}>Игра окончена!</p>
      <p className={styles.gameEndText}>
        Количество набранных очков: {currentScore}
      </p>
      <div className={styles.gameEndButtons}>
        <Link
          to="/"
          className={`${styles.linkButton} ${styles.gameEndButton}`}>
          На главную
        </Link>
        <Button onClick={restartGame} className={styles.gameEndButton}>
          Сыграть ещё раз
        </Button>
        <Link
          to="/leaderboard"
          className={`${styles.linkButton} ${styles.gameEndButton}`}>
          Посмотреть таблицу лидеров
        </Link>
      </div>
    </div>
  )
}
