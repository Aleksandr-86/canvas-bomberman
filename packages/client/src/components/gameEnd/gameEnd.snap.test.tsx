import * as renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { GameEnd } from './gameEnd'
import configureStore from 'redux-mock-store'
import { MemoryRouter } from 'react-router-dom'

const mockStore = configureStore([])
const store = mockStore({ game: { currentScore: 100 } })

describe(`Snapshot test component GameEnd`, () => {
  test(`Should correct render (with store)`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter initialEntries={[`/game`]}>
            <GameEnd onReloadGame={() => undefined} />
          </MemoryRouter>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
