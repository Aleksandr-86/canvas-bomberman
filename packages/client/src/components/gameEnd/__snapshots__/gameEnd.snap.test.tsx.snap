// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameEnd/> snapshot test cases <GameEnd/> should render <div className=gameEnd> parent with child elements 1`] = `
<div
  className="gameEnd"
>
  <p
    className="gameEndText"
  >
    Игра окончена!
  </p>
  <p
    className="gameEndText"
  >
    Количество набранных очков: 
    100
  </p>
  <div
    className="gameEndButtons"
  >
    <a
      className="linkButton gameEndButton"
      href="/"
      onClick={[Function]}
    >
      На главную
    </a>
    <button
      className="button gameEndButton"
      onClick={[Function]}
    >
      Сыграть ещё раз
    </button>
    <a
      className="linkButton gameEndButton"
      href="/leaderboard"
      onClick={[Function]}
    >
      Посмотреть таблицу лидеров
    </a>
  </div>
</div>
`;
